src/TraceEvent/Computers/StartStopActivityComputer.cs

```diff
@@ -957,53 +962,117 @@ void FixAndProcessAdoNetEvents(TraceEvent data)
-
-                // In the converter from 'DiagnosticSource' to ETW, we have some default configurations.
-                // By default, SQL (EntityFramework) DiagnosticSource event will be converted to 'Activity2Start' and 'Activity2Stop' events
-                // of provider 'Microsoft-Diagnostics-DiagnosticSource'. Meanwhile, the sql command will be in the payload.
-                // The following code is used to make the call stack looks better (converting 'Activity2Start' to 'SQLCommand')
-                // and showing the SQL command.
-                // Details:
-                //https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs
-
-                
-                var args = data.PayloadByName("Arguments") as DynamicTraceEventData.StructValue[];
-                if (args != null)
                 {
-                    var sb = Utilities.StringBuilderCache.Acquire(64);
-                    bool first = true;
-                    foreach (var arg in args)
+                    result = false;
+                }
+                else
+                {
+                    string extraInfo = null;
+                    switch (taskName)
                     {
-                        string key = arg["Key"] as string;
-                        object value = arg["Value"] as string;
-                        if (key != null && value != null)
-                        {
-                            string valueStr = value.ToString();
-                            if (!string.IsNullOrEmpty(valueStr))
+                        case "Microsoft.EntityFrameworkCore.BeforeExecuteCommand":
+                            // In the converter from 'DiagnosticSource' to ETW, we have some default configurations.
+                            // By default, SQL (EntityFramework) DiagnosticSource event will be converted to 'Activity2Start' and 'Activity2Stop' events
+                            // of provider 'Microsoft-Diagnostics-DiagnosticSource'. Meanwhile, the sql command will be in the payload.
+                            // The following code is used to make the call stack looks better (converting 'Activity2Start' to 'SQLCommand')
+                            // and showing the SQL command.
+                            // Details:
+                            //https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs
+
+                            if (0 < data.PayloadNames.Length)
                             {
-                                if (!first)
-                                    sb.Append(' ');
-                                sb.Append(key).Append("=").Append('"').Append(valueStr).Append('"');
+                                var args = data.PayloadByName("Arguments") as DynamicTraceEventData.StructValue[];
+                                if (args != null)
```

Comment:
Make a method DiagnosticSourceArgsToString(StructValue[] which encapulates this.  body here.  

I notice that the old code did this transformation for all DiagnosticSources, and you do it for only one particular one (BeforeExecuteCommand) Is that intentional?

Reply:
Seems a regression for .NET Core trace introduced by commit 5a7b2a666998b54618a09b6b10707dc8eac1d26b in PerfView. I'll fix it.